pykafka/balancedconsumer.py

```diff
@@ -547,6 +561,10 @@ def _rebalance(self):
                         raise
                     log.info('Unable to acquire partition %s. Retrying', ex.partition)
                     time.sleep(i * (self._rebalance_backoff_ms / 1000))
+            if self._partitions_assigned_callback is not None:
+                self._partitions_assigned_callback(self, new_partitions)
+            if self._partitions_revoked_callback is not None:
+                self._partitions_revoked_callback(self, current_zk_parts - new_partitions)
```

Comment:
It's called with the set of partitions that were owned and aren't anymore. The offset for each partition will be the same as it was before the rebalance. Using `held_offsets` would involve figuring out which partitions were revoked and then filtering those out of the output of `held_offsets` - a two-step process - right?